src/mip/mip_logger.h

```diff
@@ -0,0 +1,200 @@
+#pragma once
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+namespace mip {
+namespace C {
+extern "C" {
+#endif
+
+////////////////////////////////////////////////////////////////////////////////
+///@addtogroup mip_c
+///@{
+////////////////////////////////////////////////////////////////////////////////
+///@defgroup mip_logger  Mip Logging [C]
+///
+///@brief High-level C functions for logging information from within the MIP SDK
+///
+/// This module contains functions that allow the MIP SDK to log information
+/// and allows users to override the logging functions
+///
+///@{
+
+////////////////////////////////////////////////////////////////////////////////
+///@brief Logging level enum
+///
+typedef enum mip_logger_level
+{
+    MIP_LOGGER_LEVEL_OFF   = 0,  ///< Signifies that the log is turned off
```

Comment:
Nitpick, but let's shorten LOGGER to just LOG for these constants: MIP_LOG_LEVEL_XXX